src/rust/engine/options/src/config.rs

```diff
@@ -40,36 +123,80 @@ impl Config {
                 e
             )
         })?;
-        if !config.is_table() {
-            return Err(format!(
+
+        fn add_section_to_interpolation_map(
+            imap: &InterpolationMap,
+            section: Option<&Value>,
+        ) -> Result<InterpolationMap, String> {
+            let mut imap = imap.clone();
+            if let Some(section) = section {
+                if let Some(table) = section.as_table() {
+                    for (key, value) in table.iter() {
+                        if let Value::String(s) = value {
+                            imap.insert(key.clone(), s.clone());
+                        }
+                    }
+                }
+            }
+            Ok(imap)
+        }
+
+        let default_imap = add_section_to_interpolation_map(seed_values, config.get("DEFAULT"))?;
+
+        let new_sections: Result<Vec<(String, Value)>, String> = match config {
```

Comment:
Eagerly interpolating the entire config is one option: another would be to do it lazily at read time for config values. I don't have an opinion on which way to go about it, but if you had a particular reason for doing it this way, can you add a comment somewhere explaining why we do it eagerly?

Reply:
I considered doing this lazily, but it would have been a bit more clunky, and since we will almost certainly read every value in the config file at least once, there didn't seem to be any benefit to a lazy implementation.